src/server/controllers/votes.js

```diff
@@ -13,6 +25,24 @@ async function create(req, res, next) {
   }
 }
 
+async function read(req, res, next) {
+  const query = `{
+    answers
+    (where:
+      {question: "${req.params.question.toLowerCase()}"}
+    ) { id voteTokens votePower votes active question { id } }
+  }`;
+  try {
+    req.locals = req.locals || {};
```

Comment:
Looks like you're checking already in line 61 of  `src/server/controllers/index.js` if this is set, so there should be no need to set it here.

Reply:
It can't assign to `req.locals.graphData` if `req.locals` doesn't exist yet tho

Reply:
Ok, haven't seen that line! Aha!

Some ideas:

`locals` is needed when you want to transport data between middlewares, which is only halfly done here. Think this is also why I got confused why you need it here. How about making a GraphQL "prefetch" middleware and then call them after each other (first get the graphQL request) and then pass it over to `readAll`?

Also one could think about writing another "readAllFiltered" method in the base controller which takes the special graphql options and calls another custom `filterResponseFieldsAll`?